File: Java/DesignPatterns/src/main/java/designPatterns/behavioral/interpreter/MainInterpreter.java
package designPatterns.behavioral.interpreter;

public class MainInterpreter {
  public static void main(String[] args) {
    Expression isMale = new ConcreteExpression.TerminalExpression("John");
    Expression isMarried = new ConcreteExpression.TerminalExpression("Married");
    Expression isMarriedAndMale = new ConcreteExpression.AndExpression(isMale, isMarried);

    System.out.println(isMale.interpret("John"));
    System.out.println(isMarried.interpret("Married"));
    System.out.println(isMarriedAndMale.interpret("John Married"));
  }
}

File: Kotlin/DesignPatterns/src/main/kotlin/behavioral/interpreter/MainInterpreter.kt
package behavioral.interpreter

fun main() {
  var isMale = ConcreteExpression.TerminalExpression("John")
  var isMarried = ConcreteExpression.TerminalExpression("Married")
  var isMarriedAndMale = ConcreteExpression.AndExpression(isMale, isMarried)

  println(isMale.interpret("John"))
  println(isMarried.interpret("Married"))
  println(isMarriedAndMale.interpret("John Married"))
}